dateparser/parser.py

```diff
@@ -447,6 +448,19 @@ def _correct_for_time_frame(self, dateobj):
 
             dateobj = dateobj + delta
 
+        # NOTE: If this assert fires, self.now needs to be made offset-aware in a similar
+        # way that dateobj is temporarily made offset-aware.
+        assert not (self.now.tzinfo is None and dateobj.tzinfo is not None), "Review comment for details."
```

Comment:
@SimplicityGuy 
I know that it doesn't seem to happen, but Imagine we introduce the bug without noticing and some user gets this error. I think we should do the error message a bit more self-explanatory. What about something like...?:
```suggestion
        assert not (self.now.tzinfo is None and dateobj.tzinfo is not None), "`self.now` shouldn't have tzinfo. Review comment in code for details."
```

or
 
```suggestion
        assert not (self.now.tzinfo is None and dateobj.tzinfo is not None), "Timezone info is wrong. Review comment in code for details."
```

In that way, people affected by this (either developers or users) will be able to provide more context if they find this error.

@Gallaecio let me know what you think